Gems/Atom/RPI/Code/Tests/ShaderResourceGroup/ShaderResourceGroupConstantBufferTests.cpp

```diff
@@ -265,7 +265,9 @@ namespace UnitTest
             EXPECT_TRUE(m_srg->SetConstantArray<bool>(inputIndex, AZStd::array<bool, 2>({ asBools[1], asBools[2] })));
             AZStd::array_view<uint8_t> result = m_srg->GetConstantRaw(inputIndex);
             AZStd::array_view <uint32_t> resultInUint = AZStd::array_view<uint32_t>(reinterpret_cast<const uint32_t*>(result.data()), 2);
-            ExpectEqual<uint32_t>({ 1 /*true*/, 0 /*false*/ }, resultInUint);
+            EXPECT_EQ(resultInUint.size(), 2);
+            EXPECT_TRUE(resultInUint[0]);
+            EXPECT_FALSE(resultInUint[1]);
```

Comment:
Use `EXPECT_THAT` to verify containers. It will automatically verify that the sizes of the containers match, and gives really good diagnostics when they don't.
```suggestion
            EXPECT_THAT(resultInUint, testing::ElementsAre(testing::IsTrue(), testing::IsFalse()));
```
Sample output:
```
ShaderResourceGroupConstantBufferTests.cpp:263: Failure
Value of: resultInUint
Expected: has 2 elements where
element #0 is true,
element #1 is false
  Actual: { 0, 0 }, whose element #0 doesn't match
```